# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmCode.cpp
# Commit: 5ab04623c826
# Full Hash: 5ab04623c826b7f4d125fbb936754dd29af0a472
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2022-03-18 16:59:23
# Regressor Bug: 1756951
# File Overlap Count: 2
# Description:
#   Bug 1756951 - Part 1: Do not patch code when debugging. r=yury
#   
#   Instead of emitting a patchable call at entry/exit/breakpoints, emit a
#   conditional call to a per-instance handler that the debugger can
#   install.  The new code is larger than the code it replaces but within
# ==============================================================================

diff -r d209d5f4d95e -r 5ab04623c826 js/src/wasm/WasmCode.cpp
--- a/js/src/wasm/WasmCode.cpp	Fri Mar 18 07:20:52 2022 +0000
+++ b/js/src/wasm/WasmCode.cpp	Fri Mar 18 07:56:07 2022 +0000
@@ -568,7 +568,6 @@
   cursor = trapSites.serialize(cursor);
   cursor = SerializeVector(cursor, funcImports);
   cursor = SerializeVector(cursor, funcExports);
-  MOZ_ASSERT(debugTrapFarJumpOffsets.empty());
   return cursor;
 }
 
@@ -582,7 +581,6 @@
       (cursor = trapSites.deserialize(cursor)) &&
       (cursor = DeserializeVector(cursor, &funcImports)) &&
       (cursor = DeserializeVector(cursor, &funcExports));
-  MOZ_ASSERT(debugTrapFarJumpOffsets.empty());
   return cursor;
 }
 
@@ -938,9 +936,6 @@
   if (!callSites.appendAll(src.callSites)) {
     return false;
   }
-  if (!debugTrapFarJumpOffsets.appendAll(src.debugTrapFarJumpOffsets)) {
-    return false;
-  }
 #ifdef ENABLE_WASM_EXCEPTIONS
   if (!tryNotes.appendAll(src.tryNotes)) {
     return false;